dn.go

```diff
@@ -268,3 +268,15 @@ func (r *RelativeDN) hasAllAttributesFold(attrs []*AttributeTypeAndValue) bool {
 func (a *AttributeTypeAndValue) EqualFold(other *AttributeTypeAndValue) bool {
 	return strings.EqualFold(a.Type, other.Type) && strings.EqualFold(a.Value, other.Value)
 }
+
+func appendDN(baseDN, rootDN string) string {
```

Comment:
`baseDN` isn't appended by rootDN if it isn't empty. 
```go
appendDN("ou=base", "dc=test,dc=org")
```
would result in `,dc=test,dc=org`, which in turn is a invalid DN syntax. 

Also, may you add some documentation what this function is exactly for and what baseDN and rootDN stand for